### DISCUSSION TITLE
Auto-import in vitest while testing a composable throw an error

### DISCUSSION THREAD

**Discussion by Lyokolux** [nuxt/nuxt#16835] [Questions]
I'm having a hard time testing a composable in `~/composables` that auto-imports `vue` reactivity functions.
Vitest returns the error `ReferenceError: ref is not defined` (replace `ref` with whatever function of vue reactivity system that is auto-imported).

I don't know if I do something wrong (that I did not find in the documentation or online), or if it is a bug.
Even if it is a wanted behavior, it would be great to have a 0 config experience 🚀 

Here is a minimal reproduction of what I want to achieve: https://stackblitz.com/edit/nuxt-starter-geaetr?file=composables%2FmyComposable.ts,myComposable.spec.ts,vitest.config.ts

**Comment by stefffens** [comment]
I think that i have found a solution for it using [unplugin-auto-import/vite'](https://github.com/antfu/unplugin-auto-import).
i posted it in more detail here: https://stackoverflow.com/questions/72192874/testing-pinia-store-inside-nuxt3-with-vitest-throws-useruntimeconfig-not-defin/73374480#73374480

**Reply by moh1434** [reply]
In your answer, you have `setupFiles: './setupTests.ts',`.
What should I put inside it?

I instead tried:
```
test:{
    setupFiles: ["./.nuxt/imports.d.ts"],
  },
  resolve: {
    alias: {
      "~~": path.resolve(__dirname, "."),
      "#app": path.resolve(
        __dirname,
        "./node_modules/nuxt/dist/app/index.d.ts"
      ),
      "#head": path.resolve(
        __dirname,
        "./node_modules/nuxt/dist/app/index.d.ts"
      ),
      "#imports": path.resolve(__dirname, "./.nuxt/imports.d.ts"),
    },

```
but the result is: `TypeError: Package import specifier "#build/app.config.mjs" is not defined in package`

**Reply by Lyokolux** [reply]
> In your answer, you have `setupFiles: './setupTests.ts',`. What should I put inside it?
> 
> I instead tried:
> 
> ```
> test:{
>     setupFiles: ["./.nuxt/imports.d.ts"],
>   },
>   resolve: {
>     alias: {
>       "~~": path.resolve(__dirname, "."),
>       "#app": path.resolve(
>         __dirname,
>         "./node_modules/nuxt/dist/app/index.d.ts"
>       ),
>       "#head": path.resolve(
>         __dirname,
>         "./node_modules/nuxt/dist/app/index.d.ts"
>       ),
>       "#imports": path.resolve(__dirname, "./.nuxt/imports.d.ts"),
>     },
> ```
> 
> but the result is: `TypeError: Package import specifier "#build/app.config.mjs" is not defined in package`

This is currently a known issue that will be fixed: https://github.com/nuxt/framework/issues/7244
